### ISSUE TITLE
[QUESTION] Not with Apollo GraphQL

### ISSUE THREAD

**Issue by joshuaalpuerto** [alidcastano/rogue.js#54]
Can I still use this even I don't use Apollo GraphQL? Im using React Router 4 to manage my routes and I would like to explore doing SSR.

Cheers!

**Comment by alidcastano** [comment]
How do you plan to query data?

**Comment by joshuaalpuerto** [comment]
I'll be requesting the data to an API server (Rest).

**Comment by alidcastano** [comment]
The reason we recommend Graphql is that we dont call getInitialProps in pages like nextjs does.
So you ideally handle per component logic with other tools, such as querying data with graphql and redirecting with react router.

The previous version of Rogue did walk component tree and use same logic as Router to call getInitialProps in pages. But I realized it was unnecessary since all props were only being passed to App.js component (as that's the only one we have direct access to). Implementing it like Nextjs does would require wrapping React Routers components so we can pass each page its respective props, but I tried to stay away from having to do that unless I see a real need for that added complexity.
